compiler/optimizer/ValuePropagation.cpp

```diff
@@ -7739,11 +7739,21 @@ void TR::ValuePropagation::doDelayedTransformations()
       TR::TreeTop * firstTT = predictedCatchBlock->getFirstRealTreeTop();
 
       // Find the first real tree in catch block; ignoring the
-      // profiling trees possibly inserted by catch block profiler
-      //
-      while (firstTT &&
-             firstTT->getNode()->isProfilingCode())
-        firstTT = firstTT->getNextRealTreeTop();
+      // profiling trees possibly inserted by catch block profiler or
+      // asyncchecks for OSR
+      TR::TreeTop * asyncCheck = NULL;
+      while (firstTT)
+         {
+         if (firstTT->getNode()->isProfilingCode())
+            firstTT = firstTT->getNextRealTreeTop();
+         else if (firstTT->getNode()->getOpCodeValue() == TR::asynccheck)
+            {
+            asyncCheck = firstTT;
+            firstTT = firstTT->getNextRealTreeTop();
```

Comment:
This tracks an async check tree for duplication later on, but it also skips ahead to the next tree top for an async check tree whereas this was not being done in the original loop. Is this skipping for an async check tree intentional ?